index.js

```diff
@@ -0,0 +1,649 @@
+    cost = new BigNumber(self._getIssuanceCost(args))
+
+    txb.addInput(args.financeOutputTxid, args.financeOutput.n)
+    if (args.flags && args.flags.injectPreviousOutput) {
+      var chunks = bitcoinjs.script.decompile(new Buffer(args.financeOutput.scriptPubKey.hex, 'hex'))
+      txb.tx.ins[txb.tx.ins.length - 1].script = bitcoinjs.script.compile(chunks)
+    }
+
+    assetId = self._encodeAssetId(
+      args.reissueable,
+      args.financeOutputTxid,
+      args.financeOutput.n,
+      args.financeOutput.scriptPubKey.hex,
+      args.divisibility,
+      args.aggregationPolicy)
+
+    return {txb: txb, args: args, change: current - cost, assetId: assetId, totalInputs: {amount: current}}
+  }
+
+  // add to transaction enough inputs so we can cover the cost
+  // send change if any back to us
+  current = new BigNumber(0)
+  cost = new BigNumber(self._getIssuanceCost(args))
+  var change = new BigNumber(0)
+  var hasEnoughEquity = utxos.some(function (utxo) {
+    if (!isInputInTx(txb.tx, utxo.txid, utxo.index) && !(utxo.assets && utxo.assets.length)) {
+      console.log('current amount ' + utxo.value + ' needed ' + cost)
+      console.log('utxo.txid', utxo.txid)
+      console.log('utxo.index', utxo.index)
+      txb.addInput(utxo.txid, utxo.index)
+      if (txb.tx.ins.length === 1) { // encode asset
+        console.log(txb.tx.ins[0].script)
+        assetId = self._encodeAssetId(
+          args.reissueable,
+          utxo.txid,
+          utxo.index,
+          utxo.scriptPubKey.hex,
+          args.divisibility,
+          args.aggregationPolicy)
+      }
+      console.log('math: ' + current.toNumber() + ' ' + utxo.value)
+      current = current.add(utxo.value)
+      if (args.flags && args.flags.injectPreviousOutput) {
+        var chunks = bitcoinjs.script.decompile(new Buffer(utxo.scriptPubKey.hex, 'hex'))
+        txb.tx.ins[txb.tx.ins.length - 1].script = bitcoinjs.script.compile(chunks)
+      }
+      console.log('current amount: ' + current + ' projected cost: ' + cost + ' are were there yet: ' + (current.comparedTo(cost) >= 0))
+    } else {
+      console.log('skipping utxo for input, asset found in utxo: ' + utxo.txid + ':' + utxo.index)
+    }
+    return current.comparedTo(cost) >= 0
+  })
+  console.log('hasEnoughEquity: ' + hasEnoughEquity)
+  if (!hasEnoughEquity) {
+    return {success: false}
+  }
+
+  change = current - cost
+  console.log('finished adding inputs to tx')
+  console.log('change ' + change)
+  return {success: true, txb: txb, change: change, assetId: assetId, totalInputs: { amount: current }}
+}
+
+ColoredCoinsBuilder.prototype._getIssuanceCost = function (args) {
+  var self = this
+  var fee = args.fee || self.defaultFee
+  var totalCost = fee
+  console.log('_getTotalIssuenceCost: fee =', fee)
+  if (args.transfer && args.transfer.length) {
+    args.transfer.forEach(function (to) {
+      totalCost += self.mindustvalue
+    })
+  }
+
+  // TODO: calculate multisig only if actually needed
```

Comment:
This calculation is wrong... `rules` is currently not supported, and there should be an actual check for multi-sig, not necessarily when there's metadata